Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -260,23 +260,32 @@ In order to address these attack vectors, the outcome of the following checks sh
 
 #### Static Analysis
 
--- TODO [Describe how to assess this given either the source code or installer package (APK/IPA/etc.), but without running the app. Tailor this to the general situation (e.g., in some situations, having the decompiled classes is just as good as having the original source, in others it might make a bigger difference). If required, include a subsection about how to test with or without the original sources.] --
+A security concern related with protocol handlers, should be arised, when the URL is not validated or the user is not prompted for confirmation in the application before making a particular action. 
+The first step is to find out if an application registers any protocol handlers. This information can be found in `info.plist` file in the application sandbox folder. To view registered protocol handlers, simply open a project in Xcode, go to `Info` tab and open `URL Types` section, as it is presented on a below screenshot.
 
--- TODO [Confirm purpose of remark "Use the &lt;sup&gt; tag to reference external sources, e.g. Meyer's recipe for tomato soup<sup>[1]</sup>."] --
+![Document Overview](Images/Chapters/0x06h/URL_scheme.png)
 
--- TODO [Add content on static analysis of "Testing WebView Protocol Handlers" with source code) --
+Then, you should verify how an URL path is built and validated. A method responsible for handling user's URLs is called `openURL`<sup>[4]</sup>. Look for implemented controls - how an URL is validated (what input it accepts) and does performing action requires user's acceptance? 
 
 #### Dynamic Analysis
 
--- TODO [Describe how to test for this issue "Testing WebView Protocol Handlers" by running and interacting with the app. This can include everything from simply monitoring network traffic or aspects of the app’s behavior to code injection, debugging, instrumentation, etc.] --
+In a compiled applciation, you can find registered protocol handlers in a `Info.plist` file under the `CFBundleURLTypes` and then under `CFBundleURLSchemes` key. To find out an URL structure, you can simply use `strings` or `Hooper` program, e.g.
+
+```sh
+$ strings <yourapp> | grep "myURLscheme://"
+```
+> Please note that you should firstly decrypt an application (e.g. using Clutch<sup>[5]</sup>) if you want to read a binary.
+
+Once you know, what an URL structure is, you should try fuzzing an URL to force an application to perform some malicious action. The payload to your fuzzer (e.g. Burp Intruder<sup>[6]</sup>) depends on the context of protocol handler.
```

Comment:
@xep624: Do you have more information on how to fuzz the URL?
Otherwise remove this section for now. There will be a fuzzing paragraph in the document later on.
https://github.com/OWASP/owasp-mstg/issues/615